widgets/blueprintCatalog/src/RepositoryCatalog.tsx

```diff
@@ -118,6 +118,36 @@ const RepositoryCatalog: FunctionComponent<RepositoryViewProps> = ({
             main_blueprint
         } = item;
 
+        const renderLinkButton = (url: string) => {
+            const isGithub = url.includes('github');
+            const isBitbucket = url.includes('bitbucket');
+            const isGitlab = url.includes('gitlab');
+
+            let icon;
+            switch (true) {
+                case isGithub:
+                    icon = 'github';
+                    break;
+                case isBitbucket:
+                    icon = 'bitbucket';
+                    break;
+                case isGitlab:
+                    icon = 'gitlab';
+                    break;
+                default:
+                    icon = 'git';
+            }
```

Comment:
IMHO switch works better for enumaration types, not for booleans.

What about shorter version with `if`s instead of `switch`:
```suggestion
            if (isGitHub) { icon = 'github'; }
            else if (isBitbucket) { icon = 'bitbucket'; }
            else if (isGitlab) { icon = 'gitlab'; }
            else { icon = 'git'; }
```
?

Reply:
If you go with Norbert's suggestion to extract it to a function, it would look even better, sth like:

```suggestion
            if (isGitHub) return 'github';
            else if (isBitbucket) return 'bitbucket';
            else if (isGitlab) return 'gitlab';
            return 'git';
```

Reply:
I think we don't need even `else` :) modified version: https://github.com/cloudify-cosmo/cloudify-stage/pull/2239/commits/90446bd8e2bf6c38d9fddd9f7a4f399f7da13aab